src/adapters/storage.ts

```diff
@@ -74,6 +79,59 @@ export function removeCustomToken(token: Token): Token[] {
   return setCustomTokens(getCustomTokens().filter((tkn) => tkn.address !== token.address));
 }
 
+// Pending txs
+function getSerializedPendingTxs(): SerializedPendingTx[] {
+  return getStorageByKey({
+    key: constants.PENDING_TXS_KEY,
+    defaultValue: [],
+    parser: z.array(serializedPendingTxParser),
+  });
+}
+
+function setSerializedPendingTxs(
+  serializedPendingTxs: SerializedPendingTx[]
+): SerializedPendingTx[] {
+  return setStorageByKey({
+    key: constants.PENDING_TXS_KEY,
+    value: serializedPendingTxs,
+  });
+}
+
+export function getPendingTxs(env: Env): PendingTx[] {
+  const serializedPendingTxs = getStorageByKey({
+    key: constants.PENDING_TXS_KEY,
+    defaultValue: [],
+    parser: z.array(serializedPendingTxParser),
+  });
+
+  return serializedPendingTxs.map((tx) => deserializePendingTx(tx, env));
+}
+
+export function setPendingTxs(pendingTxs: PendingTx[]): PendingTx[] {
+  setStorageByKey({
+    key: constants.PENDING_TXS_KEY,
+    value: pendingTxs.map(serializePendingTx),
+  });
+
+  return pendingTxs;
+}
+
+export function addPendingTx(pendingTx: PendingTx): void {
+  const serializedPendingTxs = getSerializedPendingTxs();
+  const serializedPendingTx = serializePendingTx(pendingTx);
+
+  setSerializedPendingTxs([...serializedPendingTxs, serializedPendingTx]);
+}
```

Comment:
Same as before, I think that the domain should always work with `PendingTx` but not serialized versions specific only to this module, that's why I'm returning a `void` instead of returning a `PendingTx[]` (because we are not using them).

Reply:
Since we're now in a hurry I leave this comment open for future consideration.